packages/block-editor/src/components/rich-text/index.js

```diff
@@ -331,6 +293,12 @@ export function RichTextWrapper(
 	}
 
 	const TagName = tagName;
+	let tabIndex;
```

Comment:
How about applying the changes in this fashion which makes the code comment easier to follow:
```diff
diff --git a/packages/block-editor/src/components/rich-text/index.js b/packages/block-editor/src/components/rich-text/index.js
index 413a3c3f06..cc0fa56e17 100644
--- a/packages/block-editor/src/components/rich-text/index.js
+++ b/packages/block-editor/src/components/rich-text/index.js
@@ -153,7 +153,7 @@ export function RichTextWrapper(
 	const { getSelectionStart, getSelectionEnd, getBlockRootClientId } =
 		useSelect( blockEditorStore );
 	// Disable Rich Text editing if block bindings specify that.
-	const shouldDisableEditing = useShouldDisableEditing();
+	const shouldDisableEditing = useShouldDisableEditing( identifier );
 	const { selectionChange } = useDispatch( blockEditorStore );
 	const adjustedAllowedFormats = getAllowedFormats( {
 		allowedFormats,
@@ -293,12 +293,8 @@ export function RichTextWrapper(
 	}
 
 	const TagName = tagName;
-	let tabIndex;
-	if ( shouldDisableEditing ) {
-		tabIndex = 0;
-	} else {
-		tabIndex = props.tabIndex === 0 ? null : props.tabIndex;
-	}
+	const tabIndex = props.tabIndex === 0 ? null : props.tabIndex;
+
 	return (
 		<>
 			{ isSelected && (
@@ -393,13 +389,13 @@ export function RichTextWrapper(
 					props.className,
 					'rich-text'
 				) }
-				// Setting tabIndex to 0 is unnecessary, the element is already
+				// Setting tabIndex to 0 is unnecessary, if the element is already
 				// focusable because it's contentEditable. This also fixes a
 				// Safari bug where it's not possible to Shift+Click multi
 				// select blocks when Shift Clicking into an element with
 				// tabIndex because Safari will focus the element. However,
 				// Safari will correctly ignore nested contentEditable elements.
-				tabIndex={ tabIndex }
+				tabIndex={ shouldDisableEditing ? 0 : tabIndex }
 				data-wp-block-attribute-key={ identifier }
 			/>
 		</>
```

Reply:
Makes sense  🙂 I changed it in [this commit](https://github.com/WordPress/gutenberg/pull/58958/commits/fdb55172db765d2532502606c99f7b560faa19ae).